awscli/argprocess.py

```diff
@@ -406,6 +419,20 @@ def _should_parse_as_shorthand(self, cli_argument, value):
             LOG.debug("Param %s looks like JSON, not considered for "
                       "param shorthand.", cli_argument.py_name)
             return False
+        # The second short-circuit case is if the argument is yaml
+        if isinstance(check_val, six.string_types):
+            try:
+                yaml = YAML(typ='safe')
+                result = yaml.load(check_val)
+                # Yaml parsing is only supported for yaml list documents as there are currently no
+                # known use cases for yaml map documents
+                if isinstance(result, list):
```

Comment:
Wasn't sure if there was a use case for dictionary yaml documents. Let me know if you guys want to support it.